src/main/java/org/apache/ibatis/datasource/pooled/PooledDataSource.java

```diff
@@ -328,6 +331,7 @@ public int getPoolPingConnectionsNotUsedFor() {
    * Closes all active and idle connections in the pool.
    */
   public void forceCloseAll() {
+    fairLock.lock();
```

Comment:
What's the point of adding two locks? (fairLock and synchronized)

Reply:
sorry,I missed it when I modified it, it's fixed now